toolkit/tools/pkg/imagecustomizerlib/savedconfigs.go

```diff
@@ -40,8 +40,38 @@ func (i *IsoSavedConfigs) IsValid() error {
 	return nil
 }
 
+type PxeSavedConfigs struct {
+	IsoImageBaseUrl string `yaml:"isoImageBaseUrl"`
+	IsoImageFileUrl string `yaml:"isoImageFileUrl"`
+}
+
+func (p *PxeSavedConfigs) IsValid() error {
+	if p.IsoImageBaseUrl != "" && p.IsoImageFileUrl != "" {
+		return fmt.Errorf("cannot specify both 'isoImageBaseUrl' and 'isoImageFileUrl' at the same time.")
+	}
+	err := imagecustomizerapi.IsValidPxeUrl(p.IsoImageBaseUrl)
+	if err != nil {
+		return err
+	}
+	err = imagecustomizerapi.IsValidPxeUrl(p.IsoImageFileUrl)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type OSSavedConfigs struct {
+	DracutPackageInfo DracutPackageInformation `yaml:"dracutPackage"`
```

Comment:
Interesting - just tried it. I will need to unpack the initrd, parse the os-release file, and then clean-up.

The current implementation captures the dracut version by running `rpm` while we have the rootfs expanded, and then later the information is parsed along with the rest of the saved configs.

The caching allows us to modify an iso image faster than unpacking initrd. The code in mic will not be simpler because of the additional logic (while we already have logic for saving/loading configs - so, minimal cost there).

I think your point about being distro-independent refers to the fact that we won't depend on running `rpm` - correct? But when we support other distros, we'd probably abstract the package manager and the code will be the same at a higher level.

I'm leaning towards using the cached copy instead of re-extracting it every run.